render/detailed/summary.go

```diff
@@ -53,33 +64,58 @@ func MakeColumn(id string) Column {
 type NodeSummary struct {
 	ID           string        `json:"id"`
 	Label        string        `json:"label"`
-	Linkable     bool          `json:"linkable"` // Whether this node can be linked-to
+	LabelMinor   string        `json:"label_minor"`
+	Rank         string        `json:"rank"`
+	Shape        string        `json:"shape,omitempty"`
+	Stack        bool          `json:"stack,omitempty"`
+	Linkable     bool          `json:"linkable,omitempty"` // Whether this node can be linked-to
+	Pseudo       bool          `json:"pseudo,omitempty"`
 	Metadata     []MetadataRow `json:"metadata,omitempty"`
 	DockerLabels []MetadataRow `json:"docker_labels,omitempty"`
 	Metrics      []MetricRow   `json:"metrics,omitempty"`
+	Adjacency    report.IDList `json:"adjacency,omitempty"`
 }
 
 // MakeNodeSummary summarizes a node, if possible.
-func MakeNodeSummary(n render.RenderableNode) (NodeSummary, bool) {
-	renderers := map[string]func(report.Node) NodeSummary{
+func MakeNodeSummary(n report.Node) (NodeSummary, bool) {
+	renderers := map[string]func(NodeSummary, report.Node) (NodeSummary, bool){
+		render.Pseudo:         pseudoNodeSummary,
 		report.Process:        processNodeSummary,
 		report.Container:      containerNodeSummary,
 		report.ContainerImage: containerImageNodeSummary,
 		report.Pod:            podNodeSummary,
+		report.Service:        serviceNodeSummary,
 		report.Host:           hostNodeSummary,
 	}
 	if renderer, ok := renderers[n.Topology]; ok {
-		return renderer(n.Node), true
+		if summary, ok := baseNodeSummary(NodeSummary{}, n); ok {
+			return renderer(summary, n)
+		}
 	}
 	return NodeSummary{}, false
 }
 
+// SummarizeMetrics returns a copy of the NodeSummary where the metrics are
+// replaced with their summaries
+func (n NodeSummary) SummarizeMetrics() NodeSummary {
+	cp := n.Copy()
+	for i, m := range cp.Metrics {
+		cp.Metrics[i] = m.Summary()
+	}
+	return cp
+}
+
 // Copy returns a value copy of the NodeSummary
 func (n NodeSummary) Copy() NodeSummary {
 	result := NodeSummary{
-		ID:       n.ID,
-		Label:    n.Label,
-		Linkable: n.Linkable,
+		ID:         n.ID,
+		Label:      n.Label,
+		LabelMinor: n.LabelMinor,
+		Rank:       n.Rank,
+		Shape:      n.Shape,
+		Stack:      n.Stack,
+		Linkable:   n.Linkable,
+		Adjacency:  n.Adjacency.Copy(),
 	}
 	for _, row := range n.Metadata {
 		result.Metadata = append(result.Metadata, row.Copy())
```

Comment:
I feel like we should be using [`copy()`](https://golang.org/pkg/builtin/#copy) here